be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/handlers/GetMessagesByIdResponseHandlerSuite.scala

```diff
@@ -1,99 +1,85 @@
 package ch.epfl.pop.pubsub.graph.handlers
 
-import akka.NotUsed
-import akka.actor.{Actor, ActorRef, ActorSystem, Props, Status}
-import akka.stream.scaladsl.{Flow, Sink, Source}
-import akka.testkit.{TestKit, TestProbe}
-import ch.epfl.pop.model.network.{JsonRpcResponse, ResultObject}
+import akka.actor.{ActorRef, ActorSystem, Props}
+import akka.pattern.AskableActorRef
+import akka.stream.scaladsl.{Sink, Source}
+import akka.testkit.TestKit
+import akka.util.Timeout
+import ch.epfl.pop.model.network.JsonRpcResponse
 import ch.epfl.pop.model.network.method.message.Message
-import ch.epfl.pop.model.objects.{Base64Data, Channel, DbActorNAckException, Hash}
-import ch.epfl.pop.pubsub.AskPatternConstants
-import ch.epfl.pop.pubsub.graph.{ErrorCodes, GraphMessage}
-import ch.epfl.pop.pubsub.graph.validators.RpcValidator
-import ch.epfl.pop.storage.DbActor
+import ch.epfl.pop.model.objects.Channel
+import ch.epfl.pop.pubsub.{AskPatternConstants, MessageRegistry, PubSubMediator, PublishSubscribe}
+import ch.epfl.pop.storage.{DbActor, InMemoryStorage}
+import org.scalatest.BeforeAndAfterAll
 import org.scalatest.funsuite.AnyFunSuiteLike
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.matchers.should.Matchers.{convertToAnyShouldWrapper, equal}
 
 import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+import scala.io.BufferedSource
+import scala.util.Success
 
-class GetMessagesByIdResponseHandlerSuite extends TestKit(ActorSystem("GetMessagesByIdResponseHandlerSuiteSystem")) with AnyFunSuiteLike with AskPatternConstants {
+class GetMessagesByIdResponseHandlerSuite extends TestKit(ActorSystem("GetMessagesByIdResponseHandlerSuiteSystem")) with AnyFunSuiteLike with AskPatternConstants with BeforeAndAfterAll {
```

Comment:
Depending on the time needed to do it, it could be useful to add a test to check that messages processing are retried upon failure to minimise timing failure (the test could ask to process a greetLAO message and then a createLAO message which should both succeed in the end)

Reply:
I am no sure how such can be written, it is must be doable but I unfortunately don't have the time right now...